### ISSUE TITLE
Uploading fails - "Unexpected token < in JSON at position 0"

### ISSUE THREAD

**Issue by Banekas** [nftstorage/nft.storage#1190]
Hi,

I've been uploading a bunch of files with the store method, with the same code as show in https://github.com/nftstorage/nft.storage/blob/main/packages/client/examples/node.js/store.js inside a loop.

Unfortunately at some point it just stopped working.

**Precisions**:
- upload stopped working at the 1.850th iteration (I'm trying to upload 10.000 files alongside some properties/name/description)
- each file is around 70ko
- the error is "Unexpected token < in JSON at position 0"
- I've tried switching API keys to make sure it wasn't a limitation (weight/number) issue
- NFT storage status is shown as UP (https://status.nft.storage/)

I'm feeling like I can't do much more on my end... I noticed in the status that an incident occurred a couple of days ago, preventing uploads. Could it be this again ?

Thank you for your time and consideration.
Best regards.

**Comment by pechisworks** [comment]
I am getting the same error.
After some debugging I found a more sensible error:
`<title>Access denied | api.nft.storage used Cloudflare to restrict access</title>`
`<p>The owner of this website (api.nft.storage) has banned you temporarily from accessing this website.</p>`

What is the best possible way to upload 10.000 files to NFT Storage?

**Comment by Banekas** [comment]
Wow thanks for the answer! Could you please share the code that allowed you to find this message error?

I'm surprised though as I tried to reboot my IP address and to swtich the API key! But well, that would explain a lot!

**Comment by pechisworks** [comment]
In `nft.storage\src\lib.js` on [line 143](https://github.com/nftstorage/nft.storage/blob/fd49f4be0850604cb95867b809b7b13b0102ea07/packages/client/src/lib.js#L143) throws the error.
The client is trying to parse a json file, but as a response there comes a Cloudflare Error HTML page.

For debbing purposes i just placed a 
`
const txt = await response.text();
console.log('response', txt);
`
before that line.

**Comment by Banekas** [comment]
Well done! I didn't opened the stack trace files to see what was going on under the hood...

So I actually have a very similar error to yours:
`<title>Attention Required! | Cloudflare</title>`
...
`          <div class="cf-column">
            <h2 data-translate="blocked_why_headline">Why have I been blocked?</h2>

            <p data-translate="blocked_why_detail">This website is using a security service to protect itself from online attacks. The action you just performed triggered the security solution. There are several actions that could trigger this block including submitting a certain word or phrase, a SQL command or malformed data.</p>
          </div>`

And this is a big help because a bit lower (I haven't copied it above) I can see my IP address, and it has helped me noticed that even if I had restarted my router... it was still the same IP! Thus I was operating under the assumption I had a new IP, but I was wrong!

It's not a fix, not even a good solution, but I figure if I can change my IP address and use another API key everything should work again and I'll be able to continue uploading my NFTs.
I'll let you know if it works for me. Thanks again!

**Comment by Banekas** [comment]
So, my workaround worked, but it was a bit tricky:
- don't forget to comment out/delete your log line in nft.storage\src\lib.js 😅
- clear your browser cache/cookies (just to be safe)
- obtain a new IP address (sharing your smartphone connection is a way to go if you can't do it with your router)
- login with a new email on NFT storage and create a new API key
- replace the API key in your code
- re-launch your script

For info this time it failed after 2.400 iterations, so I'll have to repeat this a few times to upload my 10.000. Well, at least I've a way to do it now, so thx for the tip @pechisworks!

Regarding this issue in a more general way, I feel like:
- if the store method is not the way to go to upload a few 1.000 files, it should be explicitly mentioned in the doc with another proposition for doing so
- the provided code samples should feature a log (/catch) allowing one to understand the underlying error way more easily

I won't close this issue myself as my suggested workaround isn't a fix, but for my part it is good enough.

**Comment by dchoi27** [comment]
Hey there - sorry about the inconvenience! #1154 should help give a clearer error message for cases like this.

Have you all considered storing your metadata and images in a directory and uploading that? It is probably the easiest workaround. Depending on the directory size you might get some issues with larger directories, but you can split things up across a few smaller directories.

@yusefnapora can we add a few lines on this in the upcoming NFT.Storage docs? Detailing that rate limits are per IP, and that the best way around is to use directories? Some of the tips that were surfaced in this issue might be helpful as well!

**Comment by Banekas** [comment]
Hey,
No big worries, as you've seen we've managed to understand the root cause (well @pechisworks did!) and to work around it.
Yes this PR #1154 will be a blessing: at least one will know what the issue is in a straightforward way 😀

To answer your question: for one I didn't really consider uploading a directory as I used the store method for my tests from the very beginning (being unaware of the potential current issue). Consequently I didn't really want to do anything else in the end, otherwise I would have had to re-test quite a few things.

A couple of line about limitations, in the doc, would indeed be greatly appreciated.
Anyway, thanks for the answer and for the service you're providing!

Btw feel free to close this issue whenever you want 👍
I'm letting it opened just to help you remember the documentation ideas.

Best regards
